src/AddTodoForm.jsx

```diff
@@ -1,26 +1,36 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import InputWithLabel from './InputWithLabel';
 
 const AddTodoForm = ({ onAddTodo }) => {
+  const [todoTitle, setTodoTitle] = useState([]);
+  const [isInvalid, setIsInvalid] = useState(true);
+
   const handleTitleChange = (event) => {
     const newTodoTitle = event.target.value;
     setTodoTitle(newTodoTitle);
   };
 
+  useEffect(() => {
+    // check for empty string
+    if (todoTitle.length && todoTitle.trim()) {
+      setIsInvalid(false);
+    } else {
+      setIsInvalid(true);
+    }
+  }, [todoTitle]);
+
   const handleAddTodo = (event) => {
     event.preventDefault();
-    onAddTodo({ title: todoTitle, id: Date.now() });
+    onAddTodo({ title: todoTitle });
     setTodoTitle('');
   };
 
-  const [todoTitle, setTodoTitle] = useState([]);
-
   return (
     <form action="submit" onSubmit={handleAddTodo}>
       <InputWithLabel todoTitle={todoTitle} handleTitleChange={handleTitleChange} >
         Title
       </InputWithLabel>
-      <button>Add</button>
+      <button disabled={isInvalid}>Add</button>
```

Comment:
not a big deal, but I would name the state as `disabled` rather than `isInvalid` if you're only using it for disabled checks

Reply:
Fixed.